livingatlas/pipelines/src/main/java/au/org/ala/pipelines/transforms/IndexRecordTransform.java

```diff
@@ -0,0 +1,633 @@
+        .put(
+            "point-0.0001",
+            getLatLongString(lat, lon, "#.####")); // is set to IGNORE in headerAttributes
+  }
+
+  static String getLatLongStringStep(Double lat, Double lon, String format, Double step) {
+    DecimalFormat df = new DecimalFormat(format);
+    // By some "strange" decision the default rounding model is HALF_EVEN
+    df.setRoundingMode(java.math.RoundingMode.HALF_UP);
+    return df.format(Math.round(lat / step) * step)
+        + ","
+        + df.format(Math.round(lon / step) * step);
+  }
+
+  /** Returns a lat,long string expression formatted to the supplied Double format */
+  static String getLatLongString(Double lat, Double lon, String format) {
+    DecimalFormat df = new DecimalFormat(format);
+    // By some "strange" decision the default rounding model is HALF_EVEN
+    df.setRoundingMode(java.math.RoundingMode.HALF_UP);
+    return df.format(lat) + "," + df.format(lon);
+  }
+
+  static void addToIndexRecord(
+      SpecificRecordBase record, IndexRecord.Builder builder, Set<String> skipKeys) {
+
+    record.getSchema().getFields().stream()
+        .filter(n -> !skipKeys.contains(n.name()))
+        .forEach(
+            f ->
+                Optional.ofNullable(record.get(f.pos()))
+                    .ifPresent(
+                        r -> {
+                          Schema schema = f.schema();
+                          Optional<Schema.Type> type =
+                              schema.getType() == UNION
+                                  ? schema.getTypes().stream()
+                                      .filter(t -> t.getType() != Schema.Type.NULL)
+                                      .findFirst()
+                                      .map(Schema::getType)
+                                  : Optional.of(schema.getType());
+                          if (r != null) {
+                            type.ifPresent(
+                                t -> {
+                                  switch (t) {
+                                    case BOOLEAN:
+                                      //
+                                      builder.getBooleans().put(f.name(), (Boolean) r);
+                                      break;
+                                    case FLOAT:
+                                      builder.getDoubles().put(f.name(), (Double) r);
+                                      break;
+                                    case DOUBLE:
+                                      builder.getDoubles().put(f.name(), (Double) r);
+                                      break;
+                                    case INT:
+                                      builder.getInts().put(f.name(), (Integer) r);
+                                      break;
+                                    case LONG:
+                                      builder.getLongs().put(f.name(), (Long) r);
+                                      break;
+                                    default:
```

Comment:
Any need for `case ARRAY` to `multiValues`?

Does this need to align with `convertIndexRecordToSolrDoc` and `convertSolrDocToIndexRecord`?

Reply:
Good point. I've aligned them now.